receiver/oktareceiver/logs.go

```diff
@@ -0,0 +1,235 @@
+		if err != nil {
+			r.logger.Error("error creating okta api request", zap.Error(err))
+			break
+		}
+
+		// add query params to the first polling request
+		if r.nextURL == "" {
+			query := req.URL.Query()
+			query.Add("since", pollTime.Add(-r.cfg.PollInterval).Format(OktaTimeFormat))
+			query.Add("limit", strconv.Itoa(oktaMaxLimit))
+			req.URL.RawQuery = query.Encode()
+		}
+
+		req.Header.Add("Authorization", "SSWS "+string(r.cfg.APIToken))
+
+		res, err := r.client.Do(req)
+		if err != nil {
+			r.logger.Error("error performing okta api request", zap.Error(err))
+			break
+		}
+
+		if res.Body != nil {
+			defer res.Body.Close()
+		}
+
+		if res.StatusCode != http.StatusOK {
+			r.logger.Error("okta logs endpoint returned non-200 statuscode: " + res.Status)
+			break
+		}
+
+		r.setNextLink(res)
+
+		body, err := io.ReadAll(res.Body)
+		if err != nil {
+			r.logger.Error("error reading response body", zap.Error(err))
+			break
+		}
+
+		var curLogs []*okta.LogEvent
+		err = json.Unmarshal(body, &curLogs)
+		if err != nil {
+			r.logger.Error("unable to unmarshal log events", zap.Error(err))
+			break
+		}
+
+		logs = append(logs, curLogs...)
+
+		if len(curLogs) < oktaMaxLimit || curLogs[len(curLogs)-1].Published.After(pollTime) {
+			break
+		}
+	}
+
+	return logs
+}
+
+func (r *oktaLogsReceiver) processLogEvents(observedTime pcommon.Timestamp, logEvents []*okta.LogEvent) plog.Logs {
+	logs := plog.NewLogs()
+
+	for _, logEvent := range logEvents {
+		resourceLogs := logs.ResourceLogs().AppendEmpty()
+		resourceLogs.ScopeLogs().AppendEmpty()
+		logRecord := resourceLogs.ScopeLogs().At(0).LogRecords().AppendEmpty()
+
+		// resource attributes
+		resourceAttributes := resourceLogs.Resource().Attributes()
+		resourceAttributes.PutStr("okta.domain", r.cfg.Domain)
+
+		// timestamps
+		logRecord.SetObservedTimestamp(observedTime)
+		timestamp := time.UnixMilli(logEvent.Published.UnixMilli())
+		logRecord.SetTimestamp(pcommon.NewTimestampFromTime(timestamp))
+
+		// body
+		logEventBytes, err := json.Marshal(logEvent)
+		if err != nil {
+			r.logger.Error("unable to marshal logEvent", zap.Error(err))
+		} else {
+			logRecord.Body().SetStr(string(logEventBytes))
+		}
+
+		// attributes
+		logRecord.Attributes().PutStr("uuid", logEvent.Uuid)
+		logRecord.Attributes().PutStr("eventType", logEvent.EventType)
+		logRecord.Attributes().PutStr("displayMessage", logEvent.DisplayMessage)
+		logRecord.Attributes().PutStr("outcome.result", logEvent.Outcome.Result)
+		logRecord.Attributes().PutStr("actor.id", logEvent.Actor.Id)
+		logRecord.Attributes().PutStr("actor.alternateId", logEvent.Actor.AlternateId)
+		logRecord.Attributes().PutStr("actor.displayName", logEvent.Actor.DisplayName)
+	}
+
+	return logs
+}
+
+func (r *oktaLogsReceiver) Shutdown(_ context.Context) error {
+	r.logger.Debug("shutting down logs receiver")
+	r.client.CloseIdleConnections()
+	if r.cancel != nil {
+		r.cancel()
+	}
+	r.wg.Wait()
```

Comment:
I think the r.cancel() call should go first
```suggestion
	if r.cancel != nil {
		r.cancel()
	}
	r.client.CloseIdleConnections()
	r.wg.Wait()
```